kblincoe/QualOpt_SE701
# Preset date and time for sending invitation emails

This pull request completes #26.

The feature added here is the ability to preset a date and time to send invitation emails.
User specifies if they want to send emails immediately or at preset time. This preset time is used to determine the delay in sending emails and then the invitation job is registered within the scheduler.

![image](https://user-images.githubusercontent.com/20148600/37801715-7f27f226-2e8c-11e8-8829-8eb2fc7a971f.png)

The emails are scheduled using Quartz.

## **Manual Testing in development**
- Create a new study (with a participant and an email you can access)
- From the study page, select **Preset Date and Time**
- Specify a time in the future: e.g. 5 mins after current time (Note: you must keep the debugging process running since the quartz server runs in that process).
- Check your email at the time you specified to verify that the email was received.

## **Input Validation**
- If a user selects a date/ time in the past, server side validation will ensure that the email is sent out immediately.
- Client side validation has not been implemented in this PR.

## **Dependencies**
- A quartz dependency was added to the pom file, there is a small chance that this may need to modified once #49 is merged in.
- Expecting unit test cases to be added by #36, likely to be some semantic conflicts from that.
